### ISSUE TITLE
Audio device movement for cpu only augmentation

### ISSUE THREAD

**Issue by DanTremonti** [descriptinc/audiotools#84]
Given that `torchaudio.functional.apply_codec` currently only supports "cpu" device [ref](https://pytorch.org/audio/main/generated/torchaudio.functional.apply_codec.html), `apply_codec` effect may need moving the waveform to "cpu"

[The below line](https://github.com/descriptinc/audiotools/blob/master/audiotools/core/effects.py#L303) of `time_stretch` effect
```py
        waveform = self._to_2d().cpu()
```
moves the signal to "cpu", but [the following line](https://github.com/descriptinc/audiotools/blob/master/audiotools/core/effects.py#L365) is from `apply_codec` effect
```py
        waveform = self._to_2d()
```

 If I'm not missing anything here, `apply_codec` effect is missing device handling.

Another small suggestion would be to have a flag that specifies if an effect requires device movement. This will help users to effortlessly decide which effects to use, without having to look into every effect's implementation.

**Comment by pseeth** [comment]
Oh, that's fair. I can look into that and roll it into the next release!

Re: device movement, I think it should be possible to add something that moves back to the original device after the effect. The only reason for an effect to have device movement is because the underlying code doesn't support the original device. So I think we should just move stuff back to the original device, and allow users to just select latency by doing `.cuda` or `.cpu` calls between effects on the signal.

**Comment by DanTremonti** [comment]
> So I think we should just move stuff back to the original device

Yes, you are right. I was implying the same as well. But, like you also mentioned,
> only reason for an effect to have device movement is because the underlying code doesn't support the original device
 
I was merely suggesting to let user know, at least through docstring, that an effect is applied on cpu (despite moving back to input device).
